webapp/src/Controller/PublicController.php

```diff
@@ -181,42 +181,8 @@ public function teamAction(Request $request, int $teamId)
      */
     public function problemsAction()
     {
-        $contest            = $this->dj->getCurrentContest(-1);
-        $showLimits         = (bool)$this->config->get('show_limits_on_team_page');
-        $defaultMemoryLimit = (int)$this->config->get('memory_limit');
-        $timeFactorDiffers  = false;
-        if ($showLimits) {
-            $timeFactorDiffers = $this->em->createQueryBuilder()
-                    ->from(Language::class, 'l')
-                    ->select('COUNT(l)')
-                    ->andWhere('l.allowSubmit = true')
-                    ->andWhere('l.timeFactor <> 1')
-                    ->getQuery()
-                    ->getSingleScalarResult() > 0;
-        }
-
-        $problems = [];
-        if ($contest && $contest->getFreezeData()->started()) {
-            $problems = $this->em->createQueryBuilder()
-                ->from(ContestProblem::class, 'cp')
-                ->join('cp.problem', 'p')
-                ->leftJoin('p.testcases', 'tc')
-                ->select('partial p.{probid,name,externalid,problemtext_type,timelimit,memlimit}', 'cp', 'SUM(tc.sample) AS numsamples')
-                ->andWhere('cp.contest = :contest')
-                ->andWhere('cp.allowSubmit = 1')
-                ->setParameter(':contest', $contest)
-                ->addOrderBy('cp.shortname')
-                ->groupBy('cp.problem')
-                ->getQuery()
-                ->getResult();
-        }
-
-        return $this->render('public/problems.html.twig', [
-            'problems' => $problems,
-            'showLimits' => $showLimits,
-            'defaultMemoryLimit' => $defaultMemoryLimit,
-            'timeFactorDiffers' => $timeFactorDiffers,
-        ]);
+        return $this->render('public/problems.html.twig',
+            $this->dj->getTwigDataForProblemsAction(-1));
```

Comment:
This `-1` argument seems a bit hacky. Can't we make `$teamid` an optional argument, or use `null` instead?

Reply:
We might want to also clean up getCurrentContest(s), but I think it's separate from this cleanup.

See
https://github.com/DOMjudge/domjudge/blob/master/webapp/src/Service/DOMJudgeService.php#L180-L181 for the `-1` case, but `null` is also allowed - I am not sure if we use that in some places.